### DISCUSSION TITLE
Creating unauthenticated user

### DISCUSSION THREAD

**Discussion by phuoccss** [supabase/supabase#1225] [Questions]
Hi, I'm currently making sharing functionality for my site. Basically, a user can invite another user to edit/view his document via email. I'm modeling this using a many-to-many table.

If the email belongs to a signed up user, it's all good since I can look up the `uid` from `auth.users`  using the email and add it to the junction table. However, for non-user, is there any way I can create an unauthenticated user in the `auth.users` table so that I can get an `uid` to insert into the junction table. Ideally, I would not have to create another table to track invites for non-user.

I tried inserting into `auth.users` directly (just id and email) but when signing up with the same email, an error is thrown.
```js
code: 500
msg: "Database error saving new user"
error_id: "f66fd8c6-52f0-462c-a843-e938c0892dba"
```
How should I approach this?

**Comment by emschwartz** [comment]
Hi @universse,

I think the recommended way of doing this would be to create a `public.users` table that includes users created by Supabase Auth and unauthenticated users you'd create through this other mechanism. The [Create a `public.users` table](https://supabase.io/docs/guides/auth#create-a-publicusers-table) section of the docs describe this as well as how to set it up so entries (with the same ID) are automatically added to this table when new users are added to `auth.users`. You could have postgres generate a UUID for those unauthenticated users (though you'd admittedly need to merge the identifiers if a previously unauthenticated user signs up).

**Reply by phuoccss** [reply]
Thank you for your response. I'm afraid I wouldn't be able to insert unauthenticated user to `public.users` table, since `id` is a foreign key, as suggested [here](https://github.com/supabase/supabase/discussions/226#discussioncomment-89148). I guess I would have to create a separate table for tracking pending invites. 😃
